test/SlowTests/Voron/Storage/FreeScratchPages.cs

```diff
@@ -20,28 +21,43 @@ public FreeScratchPages(ITestOutputHelper output) : base(output)
         }
 
         [Fact]
-        public void UncommittedTransactionShouldFreeScratchPagesThatWillBeReusedByNextTransaction()
+        public void UncommittedTransactionShouldFreeScratchPagesThatWillBeReusedFutureTransactions()
         {
             var random = new Random();
             var buffer = new byte[1024];
             random.NextBytes(buffer);
+            Options.ManualFlushing = true;
 
-            HashSet<PageFromScratchBuffer> scratchPagesOfUncommittedTransaction;
+            PageFromScratchBuffer[] scratchPagesOfUncommittedTransaction;
 
             using (var tx = Env.WriteTransaction())
             {
-                 var tree = tx.CreateTree("foo");
-               for (int i = 0; i < 10; i++)
+                var tree = tx.CreateTree("foo");
+                for (int i = 0; i < 10; i++)
                 {
                     tree.Add("items/" + i, new MemoryStream(buffer));
                 }
 
-                scratchPagesOfUncommittedTransaction = new HashSet<PageFromScratchBuffer>(tx.LowLevelTransaction.GetTransactionPages());
+                scratchPagesOfUncommittedTransaction = tx.LowLevelTransaction.GetTransactionPages().ToArray();
 
                 // tx.Commit() - intentionally not committing
             }
 
-            HashSet<PageFromScratchBuffer> scratchPagesOfCommittedTransaction;
+            using (var tx = Env.WriteTransaction())
```

Comment:
Why do we add those write transactions now?

Reply:
We used to put freed pages from rolled back transactions into the start of the queue.
That was a lot of specialized behavior for a rare scenario. Instead, we now just treat them as normal transactions.
We'll add them to the end of the queue and free them when the _next_ transaction is flushed.

Those new write transactions are there to verify that this happens.